IndoorMapsFrontend/src/components/pages/Directory.tsx

```diff
@@ -0,0 +1,63 @@
+import { Button, Group, JsonInput } from "@mantine/core";
+import React, { useEffect } from "react";
+import { graphql, usePreloadedQuery, useQueryLoader } from "react-relay";
+import type { PreloadedQuery } from 'react-relay';
+import type { DirectoryGetAllBuildingsQuery } from "./__generated__/DirectoryGetAllBuildingsQuery.graphql";
+
+type Props = {
+
+}
+const GetAllBuildings = graphql`
+    query DirectoryGetAllBuildingsQuery {
+    allBuildings {
+      id
+      title
+      description
+    }
+  }
+`
+
+const Directory = ({ }: Props) => {
+    const [
+        queryReference,
+        loadQuery,
+    ] = useQueryLoader<DirectoryGetAllBuildingsQuery>(
+        GetAllBuildings,
+    );
+
+    useEffect(() => {
+        loadQuery({});
+    }, []);
+
+    return (
+        <>
+            <React.Suspense fallback="Loading">
+                {queryReference != null
+                    ? <NameDisplay queryReference={queryReference} />
+                    : null
+                }
+            </React.Suspense>
+        </>
+    )
+}
+
+type NameDisplayProps = {
+    queryReference: PreloadedQuery<DirectoryGetAllBuildingsQuery>
+}
+
+function NameDisplay({ queryReference }: NameDisplayProps) {
+    const data = usePreloadedQuery(GetAllBuildings, queryReference);
+
+    const buildingListElements = data.allBuildings.map((building, i) => {
+        return (
+            <Group key={i}>
```

Comment:
nit: You can directly use `building.id` as unique key rather than index to get optimal performance and avoid extra re-rendering

Reply:
I fixed this in every place except one where the relay id was not visible.  See this comment for why https://github.com/Theo-s-Meta-Capstone/IndoorMaps/commit/7a8414b3246a74cc164a9bb33fca1bbada40bdd1#diff-d8e18268ca7a4212e46ceb2c30c05314ba284f28ab3c9c435ab755765b639f00